cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java

```diff
@@ -119,6 +122,26 @@ public Response restart(@QueryParam("node") final String name) {
     }
 
     @PUT
+    @Path("rollingRestart")
+    public Response rollingRestart() throws ConfigStoreException, PersistenceException{
+        Optional<Map<String, CassandraDaemonTask>> tasksOption =
+            Optional.ofNullable(state.getDaemons());
+        if (tasksOption.isPresent()) {
+            List<CassandraDaemonTask> tasks = tasksOption.get().entrySet()
+                                                .stream()
+                                                .map(task -> task.getValue())
+                                                .collect(Collectors.toList());
+            configurationManager.updateTargetConfig();
```

Comment:
Updating the target configuration in this way is not fault tolerant.  I don't see the new configuration being persisted anywhere.  So on a scheduler crash, we have inconsistent configuration targets for nodes in the cluster.  Configuration changes are not allowed via HTTP endpoints for just this reason.  Cluster target configurations are communicated by changing the environment of the Scheduler and restarting it.  This ensures a single stable target configuration for a cluster at all times.  Even if you were to persist the configuration change, that would be insufficient as you've now added a second code path for cluster configuration modification which is also a bad idea.

Have you tested restarting the Scheduler after initiating a rolling restart in this way?  It is my strong suspicion that all Tasks which have been restarted through this mechanism will be restarted again, and those not yet restarted through this mechanism will be left alone.